src/plugins/home/server/tutorials/instructions/metricbeat_instructions.ts

```diff
@@ -6,331 +6,362 @@
-          'tar xzvf metricbeat-{config.kibana.version}-darwin-x86_64.tar.gz',
-          'cd metricbeat-{config.kibana.version}-darwin-x86_64/',
-        ],
-      },
-      DEB: {
-        title: i18n.translate('home.tutorials.common.metricbeatInstructions.install.debTitle', {
-          defaultMessage: 'Download and install Metricbeat',
-        }),
-        textPre: i18n.translate('home.tutorials.common.metricbeatInstructions.install.debTextPre', {
-          defaultMessage: 'First time using Metricbeat? See the [Quick Start]({link}).',
-          values: {
-            link: '{config.docs.beats.metricbeat}/metricbeat-installation-configuration.html',
-          },
-        }),
-        commands: [
-          'curl -L -O https://artifacts.elastic.co/downloads/beats/metricbeat/metricbeat-{config.kibana.version}-amd64.deb',
-          'sudo dpkg -i metricbeat-{config.kibana.version}-amd64.deb',
-        ],
-        textPost: i18n.translate(
-          'home.tutorials.common.metricbeatInstructions.install.debTextPost',
-          {
-            defaultMessage: 'Looking for the 32-bit packages? See the [Download page]({link}).',
-            values: { link: 'https://www.elastic.co/downloads/beats/metricbeat' },
-          }
-        ),
-      },
-      RPM: {
-        title: i18n.translate('home.tutorials.common.metricbeatInstructions.install.rpmTitle', {
-          defaultMessage: 'Download and install Metricbeat',
-        }),
-        textPre: i18n.translate('home.tutorials.common.metricbeatInstructions.install.rpmTextPre', {
-          defaultMessage: 'First time using Metricbeat? See the [Quick Start]({link}).',
-          values: {
-            link: '{config.docs.beats.metricbeat}/metricbeat-installation-configuration.html',
-          },
-        }),
-        commands: [
-          'curl -L -O https://artifacts.elastic.co/downloads/beats/metricbeat/metricbeat-{config.kibana.version}-x86_64.rpm',
-          'sudo rpm -vi metricbeat-{config.kibana.version}-x86_64.rpm',
-        ],
-        textPost: i18n.translate(
-          'home.tutorials.common.metricbeatInstructions.install.debTextPost',
-          {
-            defaultMessage: 'Looking for the 32-bit packages? See the [Download page]({link}).',
-            values: { link: 'https://www.elastic.co/downloads/beats/metricbeat' },
-          }
-        ),
-      },
-      WINDOWS: {
-        title: i18n.translate('home.tutorials.common.metricbeatInstructions.install.windowsTitle', {
-          defaultMessage: 'Download and install Metricbeat',
-        }),
-        textPre: i18n.translate(
-          'home.tutorials.common.metricbeatInstructions.install.windowsTextPre',
-          {
-            defaultMessage:
-              'First time using Metricbeat? See the [Quick Start]({metricbeatLink}).\n\
+export const createMetricbeatInstructions = memoize(
+  (context: TutorialContext) => {
```

Comment:
(see the key resolver a few hundred lines below....) In practice, we'll only memoize based on the `space`, given that the `kibanaBranch` will remain the same for a given distribution. One entry in the cache is around 2kb, so it would require a whole lot of spaces (and customers accessing the home page on all of them) for the cache to be a few Mb, so I feel like it's worth given the perf gain.

We can eventually get rid of it if we decide to properly clean that tutorial thing later.